### PR TITLE
counting sort in dart

### LINKED ISSUE #2145: Counting Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [ ] Dart 
- [x] Go - @m-code12 
- [x] Java
- [x] JS
- [x] Kotlin - @sandhyabhan
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @shivamgupta57121

### PR DESCRIPTION
**Pull request by iyoushe1703**
Fixes #2145 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added implementation of counting sort algorithm in dart

#### Languages Used:
- Dart

#### Files Added:
- Counting_Sort.dart

### CODE REVIEW

File: Counting_Sort/Counting_Sort.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import 'dart:io';
+
+class CountingSort {
+  List<int> array = [];
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.dart]: 4 space indentation across the file at the start of each block or function.

File: Counting_Sort/Counting_Sort.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    //populate temporary array with zeroes
+    for (int i = 0; i < max+1; i++)
+      count[i] = 0;
+    
+    //update newly created count array with number of occurrences 
+    for (int i = 0; i < size; i++)
+      count[arr[i]]++;
+    
+    int j = 0;
+    for (int i = 0; i < max+1; i++) {
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.dart]: Space around operators.

File: Counting_Sort/Counting_Sort.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  int size = int.parse(numberOfElements);
+
+  countingSort.inputArray(countingSort.array, size);
+  countingSort.sortingLogic(countingSort.array, size);
+  
+  print('Sorted array is: ');
+  countingSort.printArray(countingSort.array);
+}
+
+// How many elements to sort?
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.dart]: Convert this to one multiline comment.

**Comment by singh-shreya6** [comment]
Are you working on this? Will close this PR if I don't see any change in upcoming days.

**Comment by singh-shreya6** [comment]
15 days of inactivity. Closing this. Please reach out if you want to work on it again.

**Comment by iyoushe1703** [comment]
@singh-shreya6 Sorry Shreya for the inactivity. I got caught up in some unforeseen circumstances. I would like to work on it again.

**Comment by iyoushe1703** [comment]
> 15 days of inactivity. Closing this. Please reach out if you want to work on it again.

Hey! Can I still work on this? I would like to complete it.

**Pull request closed without merging**